smack-experimental/src/main/java/org/jivesoftware/smackx/mam/MamManager.java

```diff
@@ -714,6 +721,25 @@ public boolean isSupported() throws NoResponseException, XMPPErrorException, Not
         return serviceDiscoveryManager.supportsFeature(archiveAddress, MamElements.NAMESPACE);
     }
 
+    public boolean isAdvancedConfigurationSupported(Jid jid) throws InterruptedException, XMPPException, SmackException {
+        DiscoverItems discoverItems = adHocCommandManager.discoverCommands(jid);
+        for (DiscoverItems.Item item : discoverItems.getItems()) {
+            if (item.getNode().equals(ADVANCED_CONFIG_NAMESPACE))
+                return true;
```

Comment:
Nit: Please use enclosing brackets around the `return true;`

Reply:
Updated. Any other changes please let me know.